proxy/envoy/config.go

```diff
@@ -549,8 +549,7 @@ func buildClusters(services []*model.Service) []Cluster {
 				LbType:           DefaultLbType,
 				ConnectTimeoutMs: DefaultTimeoutMs,
 			}
-			if port.Protocol == model.ProtocolGRPC ||
-				port.Protocol == model.ProtocolHTTP2 {
+			if port.Protocol != model.ProtocolTCP && port.Protocol != model.ProtocolUDP {
```

Comment:
Please do not use != here. We'll be extending the set of protocols, and a check like this will cause unexpected failures.